src/internal/packager/deploy.go

```diff
@@ -44,6 +45,47 @@ func Deploy() {
 		spinner.Fatalf(nil, "Unable to find the package on the local system, expected package at %s", config.DeployOptions.PackagePath)
 	}
 
+	// If packagePath has partial in the name, we need to combine the partials into a single package
+	if strings.Contains(config.DeployOptions.PackagePath, "-000.partial-tar") {
+		spinner.Updatef("Combining partial packages into a single package")
+
+		// Replace part 0 with *
+		pattern := strings.Replace(config.DeployOptions.PackagePath, "-000.partial-tar", "-*.partial-tar", 1)
+		fileList, err := filepath.Glob(pattern)
+		if err != nil {
+			spinner.Fatalf(err, "Unable to find partial packages to combine")
+		}
+
+		pkgPath := strings.Replace(config.DeployOptions.PackagePath, "-000.partial-tar", ".tar", 1)
+		pkgFile, err := os.Create(pkgPath)
+		if err != nil {
+			spinner.Fatalf(err, "Unable to create package file")
+		}
+		defer pkgFile.Close()
+
+		for _, file := range fileList {
+			// Open the file
+			f, err := os.Open(file)
+			if err != nil {
+				spinner.Fatalf(err, "Unable to open partial package %s", file)
+			}
+			defer f.Close()
+
+			// Add the file contents to the package
+			if _, err = io.Copy(pkgFile, f); err != nil {
+				spinner.Fatalf(err, "Unable to copy partial package %s into package file", file)
+			}
+		}
+
+		// Remove the partial packages to reduce disk space before extracting
```

Comment:
I was very much not expecting this when I was playing around with the branch (and should probably be documented somewhere / somewhere more noticeable)

Reply:
As discussed during office hrs, this will be verified by shasum prior to removal and is needed to not greatly increase disk pressure.